### PR TITLE
fix(fetch): listener leaks on Socket

### LINKED ISSUE #32951: MaxListenersExceededWarning when running fetch tests
```
tests/async/test_fetch_global.py::test_should_json_stringify_body_when_content_type_is_application_json[chromium-serialization2] (node:21485) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 lookup listeners added to [Socket]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at _addListener (node:events:593:17)
    at Socket.addListener (node:events:611:10)
    at Readable.on (node:internal/streams/readable:1126:35)
    at ClientRequest.<anonymous> (/Users/maxschmitt/Developer/playwright-python/playwright/driver/package/lib/server/fetch.js:396:16)
    at ClientRequest.emit (node:events:531:35)
    at tickOnSocket (node:_http_client:850:7)
    at onSocketNT (node:_http_client:913:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
(node:21485) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 secureConnect listeners added to [Socket]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at _addListener (node:events:593:17)
    at Socket.addListener (node:events:611:10)
    at Readable.on (node:internal/streams/readable:1126:35)
    at ClientRequest.<anonymous> (/Users/maxschmitt/Developer/playwright-python/playwright/driver/package/lib/server/fetch.js:402:16)
    at ClientRequest.emit (node:events:531:35)
    at tickOnSocket (node:_http_client:850:7)
    at onSocketNT (node:_http_client:913:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
```

When running the Python fetch tests: `pytest --browser chromium tests/async/test_fetch_browser_context.py tests/async/test_fetch_global.py`

Leaks here: 

```
        socket.on('lookup', () => {
          dnsLookupAt = (0, _utils.monotonicTime)();
        });
```

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes https://github.com/microsoft/playwright/issues/32951

`node:http` reuses TCP Sockets under the hood. We weren't cleaning up our listeners, leading to the `MaxListenersExceededWarning`.

This PR adds cleanup logic. It also raises the warning threshhold, so that it doesn't trigger until there's 100 concurrent requests over the same socket.

### CODE REVIEW

**Comment by Skn0tt** [comment]
@mxschmitt could you take a look and make sure that it fixes the error in our python distro?

File: packages/playwright-core/src/server/fetch.ts [hunk 1]
```diff
@@ -312,8 +312,11 @@ export abstract class APIRequestContext extends SdkObject {
 
       let securityDetails: har.SecurityDetails | undefined;
 
+      const cleanup: (() => void)[] = [];
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/fetch.ts:315]: looks like we want `eventsHelper.addEventListener` instead.

**Comment by dgozman** [comment]
I wonder what does this mean for the timings? If the socket is reused, will it get zero `connect` timings? What if sending does not start right away - should that count as `blocked` or `queued`?

**Comment by Skn0tt** [comment]
> I wonder what does this mean for the timings? If the socket is reused, will it get zero connect timings?

Good question. I'll open a separate issue for us to investigate this.

> What if sending does not start right away - should that count as blocked or queued?

That's what https://github.com/nodejs/node/blob/d2ad9b4fb6a6c4d54f14e19e0c74f1285ee5e402/doc/api/http.md#L60-L64 sounds like, yes.

File: packages/playwright-core/src/server/fetch.ts (hunk 1 above)
Comment (Skn0tt) [reply packages/playwright-core/src/server/fetch.ts]: that's a nifty helper! done in https://github.com/microsoft/playwright/pull/32956/commits/a483cab690f75424533739c97e89fc19cd251d5a

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**